doc/content/documentation/data-structures.md

```diff
@@ -0,0 +1,223 @@
+(put [1 2 3] 3 4) # Evaluates to [1 2 3 4]
+```
+
+A vector can be counted using the `count` function.
+
+```phel
+(count []) # Evaluates to 0
+(count [1 2 3]) # Evaluates to 3
+```
+
+## Maps
+
+A Map contains key-value-pairs in random order. Each possible key appears at most once in the collection. In contrast to PHP's associative arrays, Phel Maps can have any type of keys that implement the `HashableInterface` and the `EqualsInterface`.
+
+To create a map wrap the key and values in curly brackets or use the `hash-map` function.
+
+```phel
+{:key1 value1 :key2 value2} # Create a new map with two key-value-pairs
+(hash-map :key1 value1 :key2 value2) # Create a new map using the hash-map function
+```
+
+Use the `get` function to access a value by it's key
+
+```phel
+(get {:a 1 :b 2} :a) # Evaluates to 1
+(get {:a 1 :b 2} :b) # Evaluates to 2
+(get {:a 1 :b 2} :c) # Evaluates to nil
+```
+
+To add or update a key-value pair in the map use the `put` function
+
+```phel
+(put {} :a "hello") # Evaluates to {:a "hello"}
+(put {:a "foo"} :a "bar") # Evaluates to {:a "bar"}
+```
+
+A value it a map can be removed with the `unset` function
+
+```phel
+(unset {:a "foo"} :a) # Evaluates to {}
+```
+
+As in the other data structures, the `count` function can be used to count the key-value-pairs.
+
+```phel
+(count {}) # Evalutes to 0
+(count {:a "foo"}) # Evalutaes to 1
+```
+
+## Structs
+
+A Struct is a special kind of Map. It only supports a predefined number of keys and is associated to a global name. The Struct not only defines itself but also a predicate function.
+
+```phel
+(defstruct my-struct [a b c]) # Defines the struct
+(let [x (my-struct 1 2 3)] # Create a new struct
+  (my-struct? x) # Evaluates to true
+  (get x :a) # Evaluates to 1
+  (put x :a 12) # Evaluates to (my-struct 12 2 3)
+```
+
+Internally, Phel Structs are PHP classes where each key correspondence to a object property. Therefore, Structs can be faster than Maps.
+
+## Sets
+
+A Set contains unique values in random order. All types of values are allowed that implement the `HashableInterface` and the `EqualsInterface`.
+
+A new set can be created by using the `set` function
+
+```phel
+(set 1 2 3) # Creates a new set with three values
+```
+
+The `push` function can be used to add a new value to the Set.
+
+```phel
+(push (set 1 2 3) 4) # Evaluates to (set 1 2 3)
+(push (set 1 2 3) 2) # Evaluates to (set 1 2 3)
+```
+
+Similar to the Map the `unset` function can be used to remove a value from the list
+
+```phel
+(unset (set 1 2 3) 2) # Evaluates to (set 1 3)
+```
+
+Again the `count` function can be used to count the elements in the set
+
+```phel
+(count (set)) # Evaluates to 0
+(count (set 2)) # Evaluates to 1
+```
+
+Additionally, the union of a collection of sets is the set of all elements in the collection.
+
+```phel
+(union) # Evaluates to (set)
+(union (set 1 2)) # Evaluates to (set 1 2)
+(union (set 1 2) (set 0 3)) # Evaluates to (set 0 1 2 3)
```

Comment:
In a set, the order doesn't really matter, right? I mean, technically this would be evaluated to `(set 1 2 0 3)`, but you displayed ordered just to make it visually easier. Am I right, or just because the order doesn't matter at all we can display it whatever and it will be correct anyhow? (just double checking)

Reply:
I have the same question, even though, I would display the result as `(set 1 2 0 3)` to make it clear (if the values are not ordered), [if they are, include it in the documentation? 🤔 ]

Reply:
The documentation already said that it is unordered, but for better visualization I ordered them.